OptimizelySDK/Odp/OdpManager.cs

```diff
@@ -0,0 +1,296 @@
+        /// Logger used to record messages that occur within the ODP client
+        /// </summary>
+        private ILogger _logger;
+
+        /// <summary>
+        /// Update the settings being used for ODP configuration and reset/restart dependent processes
+        /// </summary>
+        /// <param name="apiKey">Public API key from ODP</param>
+        /// <param name="apiHost">Host portion of the URL of ODP</param>
+        /// <param name="segmentsToCheck">Audience segments to consider</param>
+        /// <returns>True if settings were update otherwise False</returns>
+        public bool UpdateSettings(string apiKey, string apiHost, List<string> segmentsToCheck)
+        {
+            var newConfig = new OdpConfig(apiKey, apiHost, segmentsToCheck);
+            if (_odpConfig.Equals(newConfig))
+            {
+                return false;
+            }
+
+            _odpConfig = newConfig;
+
+            // flush old events using old odp publicKey (if exists) before updating odp key.
+            // NOTE: It should be rare but possible that odp public key is changed for the same datafile (sdkKey).
+            // - Try to send all old events with the previous public key.
+            // - If it fails to flush all the old events here (network error), remaining events will be discarded.
+            EventManager.Flush();
+            EventManager.UpdateSettings(_odpConfig);
+
+            SegmentManager.ResetCache();
+            SegmentManager.UpdateSettings(_odpConfig);
+
+            return true;
+        }
+
+        /// <summary>
+        /// Attempts to fetch and return a list of a user's qualified segments.
+        /// </summary>
+        /// <param name="userId">FS User ID</param>
+        /// <param name="options">Options used during segment cache handling</param>
+        /// <returns>Qualified segments for the user from the cache or the ODP server</returns>
+        public string[] FetchQualifiedSegments(string userId, List<OdpSegmentOption> options)
+        {
+            if (SegmentManagerOrConfigNotReady())
+            {
+                _logger.Log(LogLevel.ERROR, Constants.ODP_NOT_ENABLED_MESSAGE);
+                return null;
+            }
+
+            return SegmentManager.FetchQualifiedSegments(userId, options).ToArray();
+        }
+
+        /// <summary>
+        /// Send identification event to ODP for a given full-stack User ID
+        /// </summary>
+        /// <param name="userId">User ID to send</param>
+        public void IdentifyUser(string userId)
+        {
+            if (EventManagerOrConfigNotReady())
+            {
+                _logger.Log(LogLevel.DEBUG, "ODP identify event not dispatched (ODP disabled).");
+                return;
+            }
+
+            if (!EventManager.IsStarted)
+            {
+                _logger.Log(LogLevel.DEBUG,
+                    "ODP identify event not dispatched (ODP not integrated).");
+                return;
+            }
```

Comment:
We can drop this. It should be rare and it'll be filtered out in EventManager anyway.